armsrc/lfops.c

```diff
@@ -1417,6 +1417,56 @@ void WriteT55xx(uint32_t *blockdata, uint8_t startblock, uint8_t numblocks) {
 	}
 }
 
+// Copy Paradox ID to card and setup block 0 config
+void CopyParadoxtoT55x7(uint32_t hi2, uint32_t hi, uint32_t lo, uint8_t longFMT) {
+	uint32_t data[] = {0,0,0,0,0,0,0};
+	uint8_t last_block = 0;
+
+	if (longFMT) {
+		// Ensure no more than 84 bits supplied
+		if (hi2>0xFFFFF) {
+			DbpString("Tags can only have 84 bits.");
+			return;
+		}
+		// Build the 6 data blocks for supplied 84bit ID
+		last_block = 6;
+		// load preamble (1D) & long format identifier (9E manchester encoded)
+		data[1] = 0x0F96A900 | (manchesterEncode2Bytes((hi2 >> 16) & 0xF) & 0xFF);
+		// load raw id from hi2, hi, lo to data blocks (manchester encoded)
+		data[2] = manchesterEncode2Bytes(hi2 & 0xFFFF);
+		data[3] = manchesterEncode2Bytes(hi >> 16);
+		data[4] = manchesterEncode2Bytes(hi & 0xFFFF);
+		data[5] = manchesterEncode2Bytes(lo >> 16);
+		data[6] = manchesterEncode2Bytes(lo & 0xFFFF);
+	}	else {
+		// Ensure no more than 44 bits supplied
+		if (hi>0xFFF) {
+			DbpString("Tags can only have 44 bits.");
+			return;
+		}
+		// Build the 3 data blocks for supplied 44bit ID
+		last_block = 3;
+		// load preamble
+		data[1] = 0x0F000000 | (manchesterEncode2Bytes(hi) & 0xFFFFFF);
+		data[2] = manchesterEncode2Bytes(lo >> 16);
+		data[3] = manchesterEncode2Bytes(lo & 0xFFFF);
+	}
+	// load chip config block
+	data[0] = T55x7_BITRATE_RF_50 | T55x7_MODULATION_FSK2a | last_block << T55x7_MAXBLOCK_SHIFT;
+
+	//TODO add selection of chip for Q5 or T55x7
+	// data[0] = (((50-2)/2)<<T5555_BITRATE_SHIFT) | T5555_MODULATION_FSK2 | T5555_INVERT_OUTPUT | last_block << T5555_MAXBLOCK_SHIFT;
+
+	LED_D_ON();
+	// Program the data blocks for supplied ID
+	// and the block 0 for HID format
+	WriteT55xx(data, 0, last_block+1);
+
+	LED_D_OFF();
+
+	DbpString("DONE!");
+}
+
```

Comment:
Looks indeed very similar to the following `CopyHIDtoT55x7()`. Isn't it very simple to add the preamble as parameter and use it?

Reply:
It is, but I wasn't sure if there would be a compatibility problem with doing that (e.g. if someone uses an old client with new firmware or vice versa).

If you don't see a problem with compatibility (or you have a particular way to address it, e.g. have `CopyHIDtoT55x7` and `CopyParadoxtoT55x7` both call `CopyHIDLikeToT55x7` internally or something), then I'd be happy to do that. Just let me know.

Reply:
there shouldn't be a compatibility issue if you change the definition of CopyHIDtoT55x7 and update the 2 places it is used prior to this PR (appmain.c about lines 724, 979).
there is no need to make the parameter client side.

Reply:
something like 
`void CopyHIDtoT55x7(uint32_t hi2, uint32_t hi, uint32_t lo, uint8_t longFMT, uint8_t preamble);`
then to use:
appmain.c 979:
`CopyHIDtoT55x7(c->arg[0], c->arg[1], c->arg[2], c->d.asBytes[0], 0x1D);`